BigBaseV2/src/backend/looped/weapons/no_spread.cpp

```diff
@@ -0,0 +1,51 @@
+#include "backend/looped/looped.hpp"
+#include "pointers.hpp"
+#include "CPedFactory.hpp"
+
+namespace big
+{
+	//static std::vector<std::pair<uint32_t, float>> og_spread_values{};
+	//static uint32_t prev_weapon_hash{};
+
+	//auto is_spread_value_cached(uint32_t hash)
+	//{
+	//	return std::find_if(og_spread_values.begin(), og_spread_values.end(), [hash](auto const entry)
+	//		{
+	//			return hash == entry.first;
+	//		}) != og_spread_values.end();
+	//}
+
+	//float get_og_spread_value(uint32_t hash)
+	//{
+	//	return std::find_if(og_spread_values.begin(), og_spread_values.end(), [hash](auto const entry)
+	//		{
+	//			return hash == entry.first;
+	//		})->second;
+	//}
+
+	//auto get_spread_value(uint32_t hash)
+	//{
+	//	return g->weapons.no_spread
+	//		? 0.f
+	//		: get_og_spread_value(hash);
+	//}
+
+	void looped::weapons_no_spread()
+	{
+		auto* const weapon_mgr = (*g_pointers->m_ped_factory)->m_local_ped->m_weapon_manager;
```

Comment:
Why access weapon_mgr like this?
There's a global which exposes access to the local player ped.

```cpp
auto* const weapon_mgr = g_local_player->m_weapon_manager;
```

Reply:
Additionally check if the g_local_player != nullptr, this is a pain point in both yours and my code that might crash you if you inject early and the CPed pointer isn't accessible.